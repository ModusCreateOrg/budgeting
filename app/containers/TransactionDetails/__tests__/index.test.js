import React from 'react';
import { StaticRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { TransactionDetails } from '../index';

// mock nested component
jest.mock('components/DonutChart');

const mockOutflow = {
  transaction: {
    id: 1,
    description: "Trader Joe's food",
    value: -423.34,
    categoryId: 1,
  },
  categories: {
    1: 'Groceries',
    2: 'School',
  },
  totalBalance: 4588.07,
  isPositive: false,
};

it('renders correctly - outflow', () => {
  const tree = renderer
    .create(
      <StaticRouter context={{}}>
        <TransactionDetails {...mockOutflow} />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const mockInflow = {
  transaction: {
    id: 3,
    description: 'Ebay sale - guitar',
    value: 1102,
    categoryId: 1,
  },
  categories: {
    1: 'Groceries',
    15: 'Income',
  },
  totalBalance: 6802,
  isPositive: true,
};

it('renders correctly - inflow', () => {
  const tree = renderer
    .create(
      <StaticRouter context={{}}>
        <TransactionDetails {...mockInflow} />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const invalidFlow = {
  transaction: {},
  categories: {
    1: 'Groceries',
  },
  totalBalance: 0,
  isPositive: true,
};

it('renders correctly - invalid', () => {
  const tree = renderer
    .create(
      <StaticRouter context={{}}>
        <TransactionDetails {...invalidFlow} />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
