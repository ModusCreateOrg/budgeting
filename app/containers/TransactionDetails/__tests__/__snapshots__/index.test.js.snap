// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - inflow 1`] = `
<div
  className="transactionDetails"
>
  <a
    href="/budget"
    onClick={[Function]}
  >
    Back
  </a>
  <div
    className="isPositive"
  >
    <h1>
      Ebay sale - guitar
    </h1>
    <h2>
      Groceries
    </h2>
    <h3>
      + 16.20 %
    </h3>
  </div>
</div>
`;

exports[`renders correctly - invalid 1`] = `
<div
  className="transactionDetails"
>
  <a
    href="/budget"
    onClick={[Function]}
  >
    Back
  </a>
  <p>
    No transaction found.
  </p>
</div>
`;

exports[`renders correctly - outflow 1`] = `
<div
  className="transactionDetails"
>
  <a
    href="/budget"
    onClick={[Function]}
  >
    Back
  </a>
  <div
    className="isNegative"
  >
    <h1>
      Trader Joe's food
    </h1>
    <h2>
      Groceries
    </h2>
    <h3>
      - 9.23 %
    </h3>
  </div>
</div>
`;
